packages/amplify_datastore_plugin_interface/lib/src/types/utils/parsers.dart

```diff
@@ -1,26 +1,38 @@
-/*
- * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License").
- * You may not use this file except in compliance with the License.
- * A copy of the License is located at
- *
- *  http://aws.amazon.com/apache2.0
- *
- * or in the "license" file accompanying this file. This file is distributed
- * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
- * express or implied. See the License for the specific language governing
- * permissions and limitations under the License.
- */
-
-// ignore_for_file: public_member_api_docs
-
-// only to be used internally by amplify-flutter library
-import 'package:collection/collection.dart' show IterableExtension;
-
-String? enumToString(Object? o) =>
-    o != null ? o.toString().split('.').last : null;
-
-// only to be used internally by amplify-flutter library
-T? enumFromString<T>(String? key, List<T> values) =>
-    values.firstWhereOrNull((v) => key == enumToString(v));
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// ignore_for_file: public_member_api_docs
+
+// only to be used internally by amplify-flutter library
+import 'package:collection/collection.dart' show IterableExtension;
+import 'package:flutter/foundation.dart';
+
+String? enumToString(Object? obj) {
+  if (obj == null) {
+    return null;
+  }
+
+  return describeEnum(obj);
+}
+
+// only to be used internally by amplify-flutter library
+T? enumFromString<T>(String? key, List<T> values) =>
+    values.firstWhereOrNull((v) => key == enumToString(v));
+
+/// check whether a dynamic type [value] is an enum
```

Comment:
nit/question: should this comment say "type [data]" or should the variable below be renamed "value" in line 35, or am I missing something?

Reply:
Oops mean to rename to `value`, will update